### PR TITLE
Add exports field to fix webpack errors

### PR DESCRIPTION
**Pull request by msujew**
## Description

Adds a `exports` field to direct bundlers (mainly webpack) to the correct entry point. Having that entry in there is good practice anyway, and the other `@inversifyjs/*` packages already have an `exports` field.

## Related Issue

https://github.com/inversify/InversifyJS/issues/1616

## Motivation and Context

I assume other people will have a similar issue when it comes to using webpack and CJS (i.e. `require`) of inversify.

## How Has This Been Tested?

Applying this change to my local environment has fixed the issue reported in https://github.com/inversify/InversifyJS/issues/1616.

## Types of changes

- [ ] Updated docs / Refactor code / Added a tests case (non-breaking change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated the changelog.

### CODE REVIEW

**Comment by sdirix** [comment]
I tested the change locally on my system. It successfully fixes the issue in which webpack consumed `inversify`'s esm build instead of commonjs although inversify was `require`d.

File: package.json [hunk 1]
```diff
@@ -48,6 +48,13 @@
   "license": "MIT",
   "main": "lib/inversify.js",
   "module": "es/inversify.js",
+  "exports": {
+    ".": {
+      "types": "./lib/inversify.d.ts",
```
Comment (notaphplover) [inline package.json]:
```suggestion
```
If we go this way, tsc will default to cjs types even when ESM imports are used. Is this line critical to solve the issue?

Comment (msujew) [reply package.json]: No, I just wasn't entirely sure how tsc will behave. Thank you for the explanation! I've removed the line again :)

**Review by notaphplover — APPROVED** [review]
Thank you so much :heart:

**Pull request merged**